base/abstractarray.jl

```diff
@@ -124,7 +124,7 @@ julia> length(A)
 length(t::AbstractArray) = (@_inline_meta; prod(size(t)))
 _length(A::AbstractArray) = (@_inline_meta; prod(map(unsafe_length, indices(A)))) # circumvent missing size
 _length(A) = (@_inline_meta; length(A))
-endof(a::AbstractArray) = (@_inline_meta; length(a))
+endof(a::AbstractArray) = (@_inline_meta; last(linearindices(a)))
```

Comment:
Ideally we'd want `A[begin:end]` but of course that would need parser support. I frankly don't remember the exact context this came up in. If nothing else, this will prevent a genuine bug once we decide to allow `length` and `size` again for offset arrays. (Officially we said that would happen in 0.6, but I wonder if it's still a little early. Gotta finish https://github.com/JuliaLang/julialang.github.com/pull/498 so they get more use...)